### DISCUSSION TITLE
Lock Icon

### DISCUSSION THREAD

**Discussion by ceyerkes** [twilio-labs/paste#2636] [Request: Icon] [answered]
Hello! 

I'm in the process of exploring how tables in SendGrid can be translated into Paste. Currently working on the Custom Fields page which utilizes a lock icon:
<img width="818" alt="Screen Shot 2022-08-24 at 3 08 49 PM" src="https://user-images.githubusercontent.com/87141551/186523917-883e36af-fcd1-435d-98f2-0f016b5db090.png">

Just wondering if this icon is on the roadmap, or if I should go the self-service route. Also open to exploring another way of communicating locked/reserved for items in this table as well - maybe it doesn't even really require an icon...? Maybe a badge would suffice? Open to ideas. 

Thanks!

**Comment by SiTaggart** [comment]
Hi there

Let's take the icon first. I can't recall if that particluar icon is on the roadmap, but even if it was, the fastest way to get it would be to follow the self-service contribution. It's something we highly encourage for in need icons.

As for alternative designs or suggestions, let me ask a clarifying question, and then we can get some feedback from the team:

The RESERVED badge next to the lock icon, is it directly tied to the icon? Does the lock mean "reserved", in other words? Or can the lock represent other statuses?

**Reply by ceyerkes** [reply]
Thanks Simon! If we go the icon route, I'll do self-serve. Thought it was worth questioning the icon itself while we're at it, so I appreciate the feedback. To answer your question: The badge is tied to the icon from what I can tell, and this is the tooltip:

![Screen Shot 2022-08-25 at 9 09 43 AM](https://user-images.githubusercontent.com/87141551/186702471-481aca1b-fd8f-483f-a700-7a05b747a289.png)
 
I'm in the process of auditing Marketing Campaigns, and I haven't seen this icon used in any other table/to represent any other type of info, so I think this might only exist on the Custom Fields page. Here's what the whole page looks like:

![Screen Shot 2022-08-25 at 9 12 05 AM](https://user-images.githubusercontent.com/87141551/186703036-c8df1fcc-7410-42bb-8a73-cd7836dbf3c1.png)

**Reply by SiTaggart** [reply]
Thank @ceyerkes, let me consult the team and we'll get back to you

**Comment by SiTaggart** [comment, marked as answer]
Hi there

So after discussing with the team, we have some ideas you can play with.

It seems to us that you probably have an entire table of reserved fields. Each one is not removable, so it raises the question; are the repetitive locks and badges redundant? If you provided a short piece of content above the table, and below the "Reserved Field" heading, explaining that all of these fields are reserved and not removable, do you achieve the same thing? Do you remove their need entirely?

If that's not an option, then we also think that because Reserved and the Lock icons are tied together, you probably don't need both. Just the Reserved badge will do, but you'll have to use a [Popover with a Badge button ](https://paste.twilio.design/components/popover#popoverbadgebutton) as the replacement. We don't allow tooltips on text for accessibility reasons.

Hope that all makes sense!

**Reply by ceyerkes** [reply]
Hey Simon - removing the column completely and adding a sentence below the header to explain the reserved/locked nature of this table is exactly where we're headed after reviewing this page at content office hours, but if we can't for some reason, i'll use the popover w/ badge. Thank you!
